src/index.js

```diff
@@ -1288,6 +1288,163 @@ class CloudManagerAPI {
       throw e
     })
   }
+
+  async _listContentSets (url) {
+    return this._get(url, codes.ERROR_LIST_CONTENT_SETS).then(res => {
+      return res.json()
+    }, e => {
+      throw e
+    })
+  }
+
+  async _listContentFlows (url) {
+    return this._get(url, codes.ERROR_LIST_CONTENT_FLOWS).then(res => {
+      return res.json()
+    }, e => {
+      throw e
+    })
+  }
+
+  /**
+   * Create a Content set
+   *
+   * @param {string} programId - the program id
+   * @param {string} contentSet - the body of the content set.
+   * @returns {Promise<ContentSetRepresentation>} the created content set
+   */
+  async createContentSet (programId, contentSet) {
+    const url = `/api/program/${programId}/contentSets`
+    return this._post(url, contentSet, codes.ERROR_CREATE_CONTENTSET).then(async (res) => {
+      return halfred.parse(await res.json())
+    }, e => {
+      throw e
+    })
+  }
+
+  /**
+   * Delete Content Set
+   *
+   * There are no constraints on delete.
+   *
+   * @param {string} programId - the program id
+   * @param {string} contentSetId - the content set id
+   * @returns {Promise<object>} a truthy value
+   */
+  async deleteContentSet(programId, contentSetId) {
+    const url = `/api/program/${programId}/contentSet/${contentSetId}`
+    return this._delete(url, codes.ERROR_DELETE_CONTENTSET).then(() => {
+      return {}
+    }, e => {
+      throw e
+    })
+  }
+
+  /**
+   * Get Content Set
+   *
+   * @param {string} programId - the program id
+   * @param {string} contentSetId - the content set id
+   * @returns {Promise<ContentSetRepresentation>} the content set
+   */
+  async getContentSet(programId, contentSetId) {
+    const url = `/api/program/${programId}/contentSet/${contentSetId}`
+    return this._get(url, codes.ERROR_GET_CONTENTSET).then(res => {
+      return res.json()
+    }, e => {
+      throw e
+    })
+  }
+
+  /**
+   * List Content Sets for program
+   *
+   * @param {string} programId - the program id
+   * @returns {Promise<ContentSetRepresentation[]>} list of the content sets for the program
```

Comment:
I was reading https://github.com/adobe/aio-lib-cloudmanager/blob/main/.github/CONTRIBUTING.md#coding-patterns-and-styles, and in the `JSDoc` section it says we need to generate those types.

And I suppose README.md needs a manual update.


I was looking over those PRs as examples:
- https://github.com/adobe/aio-lib-cloudmanager/commit/6f95342e6694023e8a648f2b7181fb6eb11f55c5#diff-bfe9874d239014961b1ae4e89875a6155667db834a410aaaa2ebe3cf89820556
- https://github.com/adobe/aio-lib-cloudmanager/commit/5e6e9b958db2940743a96938d02bfef73ac6bd66
- https://github.com/adobe/aio-lib-cloudmanager/commit/aea2aa3a5b452eea72dfbd839bcbb7dc78dc4353#diff-e6c4e31d7e22c52994c96dd352a2e259349cf3d60c9fd3180eee0501dae06d60

Reply:
I see that the types need to be generated.  Looks like they are generated from the public API docs, which aren't published yet, and they are generated by the bot which will run the generate script.  The generated `api.yaml` file is not checked in

The README updates look to be autogenerated by the bot (presumably after the api.yam is generated), at least that's the impression I get from the history.

So, push to github needs to be delayed, but I think the code is now in reasonable shape. I'm working on the public API doc generation.